lib/ios-log.js

```diff
@@ -0,0 +1,203 @@
+        }
+      } catch (e) {
+        logger.error(`Could not start log capture because no iOS ` +
+        `simulator logs could be found at ${logsPath}/system.log` +
+        `Logging will not be functional for this run`);
+      }
+
+      let lastModifiedLogPath = files[0];
+      let lastModifiedLogTime = await fs.stat(files[0]).mtime;
+      for (let file in files) {
+        let mtime = await fs.stat(file).mtime;
+        if (mtime > lastModifiedLogTime) {
+          lastModifiedLogPath = file;
+          lastModifiedLogTime = mtime;
+        }
+      }
+      this.proc = new SubProcess('tail', ['-f', '-n', '1', lastModifiedLogPath]);
+      await this.finishStartingLogCapture();
+    } catch (err) {
+      logger.warn('System log capture failed.');
+      return;
+    }    
+  }  
+
+  async finishStartingLogCapture () {
+    if (!this.proc) {
+      logger.errorAndThrow('Could not capture device log');
+    }
+    this.startTimer = setTimeout(function () {
+      logger.errorAndThrow('Log capture did not start in a reasonable amount of time');
+    }, START_TIMEOUT);
+    let firstLine = true;
+
+    //this.proc.on --> start --> function startDetector
+    this.proc.on('output', (stdout, stderr) => {
+      if (stdout) {
+        if (firstLine) {
+          if (stdout.substr(-1,1) === '\n') {
+              this.onOutput();
+              firstLine = false; 
+          } 
+        } else {
+          this.logRow += stdout;
+          if (stdout.substr(-1,1) === '\n') {
+            this.onOutput('');
+            this.logRow = ''; 
+          }
+        }
+      } 
+      if (stderr) {
+        this.onOutput(' STDERR');
+      }
+    });
+
+    let sd = (stdout, stderr) => {
+      if(/execvp\(\)/.test(stderr)) {
+        throw new Error('iOS log capture process failed to start');      
+      }
+    };
+
+    await this.proc.start(sd, START_TIMEOUT); 
+  }
+
+  stopCapture () {
+    logger.debug('Stopping iOS log capture');
+    if (this.proc) {
+      this.proc.kill();
+    }
+    this.proc = null; 
+  }
+
+  onOutput (prefix) {
+    if (!this.iosLogStarted) {
+      this.iosLogStarted = true;
+      this.iosLogStartTime = new Date();
+    }
+
+    let logs = this.logRow.split('\n');
+    for (let log in logs) {
+      log = log.trim();
+      if (log) {
+        if (!this.loggingModeOn) {
+          let logRowParts = log.split(/\s+/);
+          let logRowDate = new Date(Date.parse(`${this.iosLogStartTime.getFullYear()} ${logRowParts[0]} ${logRowParts[1]} ${logRowParts[2]}`));
+            if (logRowDate.isAfter(this.iosLogStartTime)) {
+            this.loggingModeOn = true;
+            }
+          }
+        if (this.loggingModeOn) {
+          let logObj = {
+            timeStamp: Date.now()
+            , level: 'ALL'
+            , message: log
+          };
+          this.logs.push(logObj);
+          this.logsSinceLastRequest.push(logObj);
+            if (this.showLogs) logger.info(`[IOS_SYSLOG_ROW ${prefix}] ${log}`);
```

Comment:
indentation; also we try to avoid one-line ifs in most situations, so let's add some braces and make it a normal-looking 3-line if statement